src/containerapp-preview/azext_containerapp_preview/commands.py

```diff
@@ -0,0 +1,40 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+from azure.cli.command_modules.appservice.commands import ex_handler_factory
+
+from ._constants import GA_CONTAINERAPP_EXTENSION_NAME
+from ._utils import (_get_or_add_extension)
+
+
+def transform_containerapp_output(app):
+    props = ['name', 'location', 'resourceGroup', 'provisioningState']
+    result = {k: app[k] for k in app if k in props}
+
+    try:
+        result['fqdn'] = app['properties']['configuration']['ingress']['fqdn']
+    except:
+        result['fqdn'] = None
+
+    return result
+
+
+def transform_containerapp_list_output(apps):
+    return [transform_containerapp_output(a) for a in apps]
+
+
+def load_command_table(self, args):
+    # Only the containerapp related commands can ask the user to install the containerapp extension with target version
+    if len(args) > 0 and args[0] == GA_CONTAINERAPP_EXTENSION_NAME:
+        if not _get_or_add_extension(self, GA_CONTAINERAPP_EXTENSION_NAME):
+            return
```

Comment:
Does the logic here seem unnecessary? Because the CLI supports dynamic load, and `load_arguments` will also determine whether the extension has been loaded

Reply:
I want to make sure that the user prompt is done before executing the command.
I think it would be more reassuring to add this code here if the user is only installing the preview and can be prompted to install GA.

But if load_arguments must be executed before load_command forever, the logic here is unnecessary.

Reply:
Update:
Need the prompt logic in load_commands, as containerapp-preview required containerapp extension.